cache_utils/decorators.py

```diff
@@ -46,16 +71,26 @@ def test(*args, **kwargs):
         cache_backend = caches['default']
 
     def _cached(func):
-        func_type = _func_type(func)
+        fn_key_str = str(
+            fn_key(func)
+            if callable(fn_key)
+            else fn_key
+        )
+
+        if not fn_key_str:
+            raise ValueError('fn_key must be non-empty')
+
+        def _get_key(*args, **kwargs):
+            args_str = serialize(key(*args, **kwargs))
+            return sanitize_memcached_key(
+                '%s%s(%s)' % (PREFIX, fn_key_str, args_str)
+            )
 
         @wraps(func)
         def wrapper(*args, **kwargs):
-            full_name(*args)
-
             # try to get the value from cache
-            key = _get_key(wrapper._full_name, func_type, args, kwargs)
+            key = _get_key(*args, **kwargs)
```

Comment:
👍 

also, makes perfect sense to move all the `_get_key` and `fn_key_str` computations inside the `_cached` context, since we have the function and can drop one param.